CMakeLists.txt

```diff
@@ -7,6 +7,14 @@
 # and loading the cross-compilation settings from CMAKE_TOOLCHAIN_FILE.
 
 cmake_minimum_required(VERSION 3.16)
+
+# https://github.com/microsoft/vcpkg#vcpkg-as-a-submodule-with-cmake
+# > When using vcpkg as a submodule of your project, 
+# > you can add the following to your CMakeLists.txt before the first project() call,
+# > instead of passing CMAKE_TOOLCHAIN_FILE to the cmake invocation.
+set(CMAKE_TOOLCHAIN_FILE "${CMAKE_CURRENT_SOURCE_DIR}/vcpkg/scripts/buildsystems/vcpkg.cmake"
```

Comment:
This change should only go in OptionsJSCOnly.cmake and when `USE_BUN_JSC_ADDITIONS` is `ON`

Reply:
But it needed to be added before the first project() call, are we going only to use vcpkg on Windows?

Reply:
We can also remove this, and set it in `CMakePresets.json` or invoking cmake with `-DCMAKE_TOOLCHAIN_FILE`